1_fetch/src/unzip_file.py

```diff
@@ -0,0 +1,33 @@
+import os
+import zipfile
+
+def unzip_file(filename, source_dir, destination_dir):
+    """
+    Unzip file, save results to another directory.
+
+    :param filename: File to unzip, with full paths
+    :param source_dir: Directory to pull zipped file from.
+        This part of the path is removed from the beginning of filename to determine where to save unzipped files.
+    :param destination_dir: Directory to save unzipped files to.
+        In each file path, source_dir is replaced with destination_dir to determine where to save unzipped files.
+    :returns: list of paths to unzipped files
+
+    """
+    # check that file is inside source_dir
+    if os.path.samefile(source_dir, os.path.commonpath([filename, source_dir])):
+        relfile = os.path.relpath(filename, source_dir)
+        destination_file = os.path.join(destination_dir, os.path.splitext(relfile)[0])
```

Comment:
I'd suggest renaming this to `destination_subdir` rather than calling it a file, since it's the directory into which the files will be unzipped, right?